src/ape/managers/chain.py

```diff
@@ -276,13 +277,16 @@ def poll_blocks(
             raise ValueError("'stop' argument must be in the future.")
 
         # Get number of last block with the necessary amount of confirmations.
-        latest_confirmed_block_number = self.height - required_confirmations
-        has_yielded_before_reorg = False
+        last_yielded_hash = None
+        last_yielded_number = None
 
         if start_block is not None:
-            # Front-load historically confirmed blocks.
-            yield from self.range(start_block, latest_confirmed_block_number + 1)
-            has_yielded_before_reorg = True
+            # Front-load historical blocks.
+            for block in self.range(start_block, self.height - required_confirmations + 1):
+                last_yielded_hash = block.hash
+                last_yielded_number = block.number
+                time_since_last = time.time()
+                yield block
```

Comment:
Maybe a little optimization trick here:

```suggestion
                yield block

            # NOTE: using last `block` from historical iterator
            last_yielded_hash = block.hash
            last_yielded_number = block.number
            time_since_last = time.time()
```

In python, `block` exists past the scope of the iterator here, and you only need the last one, so leveraging that will reduce a bunch of small computation if you're doing a large number of blocks